auth-server/src/test/java/com/makeurpicks/service/OAuthServiceTest.java

```diff
@@ -0,0 +1,199 @@
+import com.makeurpicks.dao.ClientDetailsDAO;
+import com.makeurpicks.domain.OAuthClientDetails;
+import com.makeurpicks.domain.OAuthClientDetailsBuilder;
+import com.makeurpicks.exception.OAuthclientValidationException;
+import com.makeurpicks.exception.OAuthclientValidationException.OAuthClientExceptions;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = AuthServerApplication.class)
+//@RunWith(MockitoJUnitRunner.class)
+
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+public class OAuthServiceTest {
+
+	
+	@Autowired
+	private OAuthClientsService oAuthClientService;
+	
+	
+	
+	@Mock
+	public ClientDetailsDAO clientDetailsDAO;
+	
+
+	@Rule
+	public ExpectedException expectedEx = ExpectedException.none();
+	
+	@Before
+	public void setup() {
+		MockitoAnnotations.initMocks(this);
+		
+		OAuthClientDetails authClientDetails=new OAuthClientDetails();
+		authClientDetails.setClientId("1");
+		authClientDetails.setClient_secret("111111");
+		
+		OAuthClientDetails authClientDetails2=new OAuthClientDetails();
+		authClientDetails.setClientId("2");
+		authClientDetails.setClient_secret("22222");
+		clientDetailsDAO.save(authClientDetails);
+		clientDetailsDAO.save(authClientDetails2);
+	}
+	
+	@Test
+	public void testGetAllClients(){
+		if(oAuthClientService==null){
+			fail();
+		}
+		Iterable<OAuthClientDetails > clients=oAuthClientService.findAllClients();
+		List<OAuthClientDetails> oauthclientdetailslist=new ArrayList<OAuthClientDetails>();
+		if(oauthclientdetailslist!=null){
+			clients.forEach(oAuthClientDetails -> oauthclientdetailslist.add(oAuthClientDetails));	
+		}
+		
+		assertNotEquals(Long.valueOf(0).longValue(), Long.valueOf(oauthclientdetailslist.size()).longValue());
+		//assertEquals(2, oauthclientdetailslist.size());
+		
+		
+	}
+	
+	
+	@Test
+	public void testCreateClientsInValidDataNoClientId() throws OAuthclientValidationException {
+		if(oAuthClientService==null){
+			fail();
+		}
+		expectedEx.expect(OAuthclientValidationException.class);
+		expectedEx.expectMessage(contains(OAuthClientExceptions.CLIENT_ID_NULL.toString()));
+		//assertThat(expectedEx.e)
+		//SINCE THERE COULD BE MANY REASONS WITHINA SINGLE EXCEPTION BLOCK, GET MESSAGES RETURN A ARRAY. COMPARING IT WITH A 
+		//SINGLE STRING ALWAYS MAKES IT FAIL. HENCE USING MATCHER
+
+		
+		//OAuthClientDetails oAuthClientDetails=new OAuthClientDetailsBuilder(null, "testsecret", "ROLE_TRUSTED_CLIENT","client_credentials,password,authorization_code,refresh_token", "read,write","http://localhost", Boolean.FALSE.toString()).build();
+		OAuthClientDetails oAuthClientDetails=new OAuthClientDetailsBuilder(null, "testsecret", "ROLE_TRUSTED_CLIENT","client_credentials,password,authorization_code,refresh_token", "read,write",null, Boolean.FALSE.toString()).build();
+
+		OAuthClientDetails output=null;
+		
+		output = oAuthClientService.createOAuthClientDetails(oAuthClientDetails);
+			
+		assertNull(output);
```

Comment:
Here the assert is redundant, I see you are expecting a exception, so you know the output otherwise is garbage.